apps/governance/src/routes/proposals/components/proposal-terms/proposal-terms.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  RoundedWrapper,
+  KeyValueTable,
+  KeyValueTableRow,
+  Icon,
+} from '@vegaprotocol/ui-toolkit';
+import { BigNumber } from '../../../../lib/bignumber';
+import { SubHeading } from '../../../../components/heading';
+import { useTranslation } from 'react-i18next';
+import { useState } from 'react';
+import classnames from 'classnames';
+
+interface ProposalTermsProps {
+  data: Record<string, unknown>;
+}
+
+const getParsedValue = (value: unknown) => {
+  if (typeof value === 'string') {
+    try {
+      // Check if value is a number string - if so use bignumber to maintain precision
+      if (/^\d+(\.\d+)?$/.test(value)) {
+        return new BigNumber(value).toString();
+      } else {
+        // This would convert, for example, a JSON object ('{"key":"value"}')
+        // into an actual JS object ({key: "value"})
+        return JSON.parse(value);
+      }
+    } catch (error) {
+      return value;
+    }
+  } else {
+    return value;
+  }
+};
+
+const renderKeyValue = (key: string, value: string | number) => (
+  <KeyValueTable>
+    <KeyValueTableRow>
+      {key}
+      {value}
+    </KeyValueTableRow>
+  </KeyValueTable>
+);
+
+const renderArray = (key: string, array: unknown[]) => {
+  if (array.every((item) => typeof item === 'string')) {
+    return renderKeyValue(key, array.join(', '));
```

Comment:
suggestion (non blocking): Isn't this just a component? Shouldn't it be upper cased and called with angle brackets?

Reply:
Yes it should - this function/component went through a couple of changes to it's functionality and that got overlooked, as it did with the others. Thanks